### PR TITLE
feat: support new format address

### PR DESCRIPTION
**Pull request by shaojunda**

### CODE REVIEW

File: spec/ckb/address_spec.rb [hunk 1]
```diff
@@ -6,7 +6,16 @@
   let(:pubkey_blake160) { "0x36c329ed630d6ce750712a477543672adab57f4c" }
   let(:pubkey_blake160_bin) { Utils.hex_to_bin(pubkey_blake160) }
   let(:prefix) { "ckt" }
-  let(:address) { "ckt1qyqrdsefa43s6m882pcj53m4gdnj4k440axqswmu83" }
+  let(:data_hash) { "0xa656f172b6b45c245307aeb5a7a37a176f002f6f22e92582c58bf7ba362e4176" }
+  let(:type_hash) { "0x1892ea40d82b53c678ff88312450bbb17e164d7a3e0a90941aa58839f56f8df2" }
+  let(:pubkey_hash160) { "0xc8045f588e627a8381810923c61d0705d10b86d3" }
+  let(:type10_address) { "ckt1qyqrdsefa43s6m882pcj53m4gdnj4k440axqswmu83" }
```
Comment (ashchan) [inline spec/ckb/address_spec.rb]: Anyone understands what is `type2`, `type4`, `type10` and `type11`? 😿

Comment (shaojunda) [reply spec/ckb/address_spec.rb]: need to read RFC
```
type2 means format_type = "0x02"
type4 means format_type = "0x04"
```
any better suggestions?

Comment (shaojunda) [reply spec/ckb/address_spec.rb]: how about `full_payload_address2` and `full_payload_address4`?

Comment (shaojunda) [reply spec/ckb/address_spec.rb]: thanks, fixed.

**Pull request merged**